src/NServiceBus.Core/Recoverability/Faults/MoveFaultsToErrorQueueBehavior.cs

```diff
@@ -71,12 +79,26 @@ public MoveFaultsToErrorQueueBehavior(CriticalError criticalError, IPipelineBase
         HostInformation hostInformation;
         BusNotifications notifications;
         string errorQueueAddress;
+        string localAddress;
         static ILog Logger = LogManager.GetLogger<MoveFaultsToErrorQueueBehavior>();
 
         public class Registration : RegisterStep
         {
-            public Registration()
-                : base("MoveFaultsToErrorQueue", typeof(MoveFaultsToErrorQueueBehavior), "Moved failing messages to the configured error queue")
+            public Registration(ReadOnlySettings settings, string localAddress)
+                : base("MoveFaultsToErrorQueue", typeof(MoveFaultsToErrorQueueBehavior), "Moved failing messages to the configured error queue", b =>
+                {
```

Comment:
I'm totally OK with keeping the RegisterStep. But then we need to make the creation of the behavior a first class citizen on the RegisterStep API (so essentially it becomes the factory). Instead of using a magic Func in the constructor.